contracts/v3/strategies/MIMConvexStrategy.sol

```diff
@@ -0,0 +1,136 @@
+        IStableSwap2Pool _stableSwap2Pool,
+        address _controller,
+        address _manager,
+        address _router
+    ) public BaseStrategy(_name, _controller, _manager, _want, _weth, _router) {
+        (, address _token, , address _crvRewards, , ) = _convexVault.poolInfo(_pid);
+        crv = _crv;
+        cvx = _cvx;
+        mim = _mim;
+        cvx3 = _cvx3;
+        pid = _pid;
+        convexVault = _convexVault;
+        mimCvxDepositLP = _token;
+        crvRewards = IConvexRewards(_crvRewards);
+        stableSwap2Pool = _stableSwap2Pool;
+        // Required to overcome "Stack Too Deep" error
+        _setApprovals(
+            _want,
+            _crv,
+            _cvx,
+            _mim,
+            _cvx3,
+            address(_convexVault),
+            address(_stableSwap2Pool)
+        );
+    }
+
+    function _setApprovals(
+        address _want,
+        address _crv,
+        address _cvx,
+        address _mim,
+        address _cvx3,
+        address _convexVault,
+        address _stableSwap2Pool
+    ) internal {
+        IERC20(_want).safeApprove(address(_convexVault), type(uint256).max);
+        IERC20(_crv).safeApprove(address(router), type(uint256).max);
+        IERC20(_cvx).safeApprove(address(router), type(uint256).max);
+        IERC20(_mim).safeApprove(address(_stableSwap2Pool), type(uint256).max);
+        IERC20(_cvx3).safeApprove(address(_stableSwap2Pool), type(uint256).max);
+        IERC20(_want).safeApprove(address(_stableSwap2Pool), type(uint256).max);
+    }
+
+    function _deposit() internal override {
+        convexVault.depositAll(pid, true);
+    }
+
+    function _claimReward() internal {
+        crvRewards.getReward(address(this), true);
+    }
+
+    function _addLiquidity() internal {
+        uint256[2] memory amounts;
+        amounts[0] = IERC20(mim).balanceOf(address(this));
+        amounts[1] = IERC20(cvx3).balanceOf(address(this));
+        stableSwap2Pool.add_liquidity(amounts, 1);
+    }
+
+    function getMostPremium() public view returns (address, uint256) {
+        uint256[] memory balances = new uint256[](2);
+        balances[0] = stableSwap2Pool.balances(0); // MIM
+        balances[1] = stableSwap2Pool.balances(1); // 3CRV
+
+        if (balances[0] > balances[1]) {
+            // MIM
+            return (cvx3, 1);
+        }
+
+        return (mim, 0); // If they're somehow equal, we just want MIM
+    }
+
+    function _harvest(uint256 _estimatedWETH, uint256 _estimatedYAXIS) internal override {
+        _claimReward();
+        uint256 _cvxBalance = IERC20(cvx).balanceOf(address(this));
+        if (_cvxBalance > 0) {
+            _swapTokens(cvx, crv, _cvxBalance, 1);
+        }
+
+        uint256 _remainingWeth = _payHarvestFees(crv, _estimatedWETH, _estimatedYAXIS);
+
+        if (_remainingWeth > 0) {
+            (address _token, ) = getMostPremium(); // stablecoin we want to convert to
+            _swapTokens(weth, _token, _remainingWeth, 1);
```

Comment:
Suggestion: consider requiring an output amount to _swapTokens (e.g. not just 1), to prevent slippage.
See: https://github.com/code-423n4/2021-09-yaxis-findings/issues/133
Note, this has been "sponsor disputed"